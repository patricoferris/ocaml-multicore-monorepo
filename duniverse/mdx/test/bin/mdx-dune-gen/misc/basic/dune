(rule
 (target dune_gen.ml)
 (action
  (with-stdout-to
   %{target}
   (run ocaml-mdx dune-gen --prelude %{dep:prelude.ml} --prelude
     %{dep:prelude2.ml}))))

(rule
 (alias runtest)
 (action
  (diff %{dep:dune.gen.expected} %{dep:dune_gen.ml})))
