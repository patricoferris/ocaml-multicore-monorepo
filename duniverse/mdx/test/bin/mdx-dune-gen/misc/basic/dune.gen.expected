let run_exn_defaults =
  let open Mdx_test in
  let packages =
    Package.[
      unix;
      findlib_top;
      findlib_internal;
      compilerlibs_toplevel;
    ]
  in
  let predicates = Predicate.[ byte; toploop ] in
  let non_deterministic =
    match Sys.getenv_opt "MDX_RUN_NON_DETERMINISTIC" with
    | Some _ -> true
    | None -> false
  in
  run_exn ~packages ~predicates ~prelude_str:[]
    ~non_deterministic
    ~silent_eval:false ~record_backtrace:false
    ~syntax:None ~silent:false
    ~verbose_findlib:false ~section:None
    ~root:None ~force_output:false
    ~output:(Some `Stdout)
let file = Sys.argv.(1)
let prelude = ["prelude.ml";"prelude2.ml"]

let directives = List.map (fun path ->
  Mdx_top.Directory path) []

let _ = run_exn_defaults
  ~file
  ~prelude
  ~directives
